src/master-playlist-controller.js

```diff
@@ -708,6 +807,91 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
+   * Determine the correct subtitle playlist based on the active
+   * SubtitleTrack and initialize a PlaylistLoader and SegmentLoader if
+   * necessary. This method is called once automatically before
+   * playback begins to enable the default subtitle track and should be
+   * invoked again if the track is changed.
+   */
+  setupSubtitles() {
+    let subtitleGroup = this.activeSubtitleGroup_();
+    let track = this.activeSubtitleTrack_();
+
+    this.subtitleSegmentLoader_.pause();
+
+    if (!track) {
+      // stop playlist and segment loading for subtitles
+      if (this.subtitlePlaylistLoader_) {
+        this.subtitlePlaylistLoader_.dispose();
+        this.subtitlePlaylistLoader_ = null;
+      }
+      return;
+    }
+
+    let properties = subtitleGroup.filter((subtitleProperties) => {
+      return subtitleProperties.id === track.id;
+    })[0];
+
+    // startup playlist and segment loaders for the enabled subtitle track
+    if (!this.subtitlePlaylistLoader_ ||
+        // if the media hasn't loaded yet, we don't have the URI to check, so it is
+        // easiest to simply recreate the playlist loader
+        !this.subtitlePlaylistLoader_.media() ||
+        this.subtitlePlaylistLoader_.media().resolvedUri !== properties.resolvedUri) {
+
+      if (this.subtitlePlaylistLoader_) {
+        this.subtitlePlaylistLoader_.dispose();
+      }
+
+      // reset the segment loader
+      this.subtitleSegmentLoader_.resetEverything();
```

Comment:
`setupSubtitles` gets called everytime the player's `TextTrackList` fires a `change` event. When changing subtitle tracks through the player ui, depending on which track was selected and which track you are changing to, this event will fire multiple times. (it also fires a change event if you select the already active track in the UI) Only one of these calls will actually reset the playlist loader and segment loader to start loading from a new track. By only resetting when we are actually changing tracks, we avoid reseting the segment loader after its already started from the previous call, or if the user selected the already active track, then all that happens is the segment loader is paused and unpaused. Also, yes it should be reset even if there is no track, but since we always reset before starting a new track, it will get reset when the user turns subtitles back on.